include/sys/mutex.h

```diff
@@ -55,11 +55,19 @@ static inline void sys_mutex_init(struct sys_mutex *mutex)
 	 */
 }
 
+#ifdef __cplusplus
```

Comment:
Why not place these guards at the beginning and end of the file, as it is customary and done in many other header files in Zephyr?

Reply:
I changed it to be at the start and end of the file (or do should I just add it for the Userspace section after the includes, since all functions in #else are static anyways)